src/engraving/libmscore/note.cpp

```diff
@@ -2133,48 +2133,138 @@ void Note::setHeadHasParentheses(bool hasParentheses)
 
 void Note::setDotY(DirectionV pos)
```

Comment:
This function looks now too overcomplicated. Too many nested conditions and loops. I would recommend moving nested code to separate private methods

Reply:
Done!